[wasm] Misc fixes

radical opened the pull request:
1. Wasm.Build.Tests started to fail when installing the workload manifests with https://github.com/dotnet/runtime/pull/76849 which moved to `Stable branding for GA`. This caused the workload installation to fail because the targets were using the incorrect manifest version - 7.0.0-ci instead of 7.0.0. It is fixed here by setting that early, and using that in all the relevant places.
  a. The `PackageVersion` is not set at project evaluation time, so instead we move setting the Version used to a new target.

3. Changes based on Viktor Hofer's review feedback in https://github.com/dotnet/runtime/pull/51327#pullrequestreview-651576417
Fixes https://github.com/dotnet/runtime/issues/73758 .

radical committed:
[wasm] Fix Wasm.Build.Tests with stable 7.0.0

Set the manifest package version to use (`7.0.0` instead of `7.0.0-ci`)
early.

Fixes https://github.com/dotnet/runtime/issues/76908 .

radical committed:
Misc fixes based on review feedback

Fixes https://github.com/dotnet/runtime/issues/73758 .

radical commented:
Best reviewed with `ignore whitespace`.

ViktorHofer commented on src/tasks/WorkloadBuildTasks/WorkloadBuildTasks.csproj:
```diff
@@ -11,9 +10,4 @@
 
     <PackageReference Include="Microsoft.Build.Tasks.Core" Version="$(MicrosoftBuildTasksCoreVersion)" />
   </ItemGroup>
-
-  <Target Name="PublishBuilder"
-          AfterTargets="Build"
-          DependsOnTargets="Publish" />
-
```
Do you remember why this existed?

radical replied:
No, but it seems like a mistake. So, I'm waiting to see how the CI run goes.

ViktorHofer replied:
Maybe also double check the nuget package content. That should be a good indicator to see if something was previously added from the publish folder.

radical replied:
This is used only for preparing a sdk+workload installation for testing only.

ViktorHofer commented on eng/testing/tests.wasm.targets:
```diff
@@ -154,13 +154,13 @@
                           ManifestName="Microsoft.NET.Workload.Mono.ToolChain.net7"
                           Variant="net7"
-                          Version="$(PackageVersion)"
+                          Version="$(PackageVersionForWorkloadManifests)"
```
If this is a required property to be set, consider adding a target that validates it and otherwise throw via the Error msbuild task.

radical replied:
If this isn't set correctly then it will fail when we try to install the workload for testing. And that emits better debug info for it too.

radical commented:
Thank you for the review!

radical committed:
Update WorkloadBuildTasksDir to remove 'publish' in the path

radical committed:
workloads-testing.targets: ignore symbols.nupkgs when looking for runtime packs

radical committed:
Fix WBT to have the correct version for WORKLOAD_PACKS_VER

radical committed:
[wasm] Emit an error if PackageVersionForWorkloadManifests property is not set

radical committed:
workloads-testing: Get the PackageVersion at build time

.. because it is not set at evaluation time. So, get it in a target, and
move the WorkloadIds to a target too.

radical committed:
Merge remote-tracking branch 'origin/main' into nit-fixes

radical commented:
The error checks @ViktorHofer suggested are useful, and so I added couple of those.

radical committed:
Ensure the version is set before generating the runscript for WBT

radical committed:
Merge remote-tracking branch 'origin/main' into nit-fixes

radical committed:
Merge branch 'dotnet:main' into nit-fixes

radical committed:
Merge branch 'dotnet:main' into nit-fixes
